chapter3.adoc

```diff
@@ -188,8 +188,21 @@ respective sections in this specification.
 The `MFENCE.SPA` fence instruction is used to synchronize updates to supervisor
 domain access-permissions with current execution.
 `MFENCE.SPA` is only valid in M-mode. If operand rs1≠x0, it
-specifies a single physical address, and if rs2≠x0, it specifies
-a single SDID. Executing a `MFENCE.SPA` guarantees that any previous stores
+specifies a single physical address, and if rs2≠x0, it specifies a single SDID. 
```

Comment:
What is the granularity of an `MFENCE.SPA` with an address? This wording suggests that the granularity is a single byte, which seems rather unlikely. Compare the description of `SFENCE.VMA` in the privileged spec (emphasis mine):

> If rs1≠x0 and rs2=x0, the fence orders only reads and writes made to **leaf page table entries corresponding to the virtual address** in rs1, for all address spaces.

Should `MFENCE.SPA` do the same? In this context, if a MTTL2 entry has a type other than MTT_L1_DIR, would a "leaf page table entry" refer to the 32 MiB covered by that single MTTL2 entry, or the 2 MiB/4 MiB/1 GiB page containing the physical address?

Reply:
Correct... we could add this statement for the cases where rs1≠x0:

_If rs1 corresponds to an `MTT_L1_DIR` mapping, the fence applies to a 4 KiB page. If rs1 corresponds to a `2M_PAGES` or `4M_PAGES` mapping, the fence applies to a 2 MiB or 4 MiB range. If rs1 corresponds to one of the `1G_*` mappings, the fence applies to a 1 GiB range._

Reply:
sgtm

Reply:
@eckhard-delfs-qualcomm please update - thanks